keras_cv/layers/mbconv.py

```diff
@@ -0,0 +1,184 @@
+        self.kernel_size = kernel_size
+        self.strides = strides
+        self.se_ratio = se_ratio
+        self.bn_momentum = bn_momentum
+        self.activation = activation
+        self.survival_probability = survival_probability
+        self.filters = self.input_filters * self.expand_ratio
+        self.filters_se = max(1, self.filters)
+
+        self.conv1 = layers.Conv2D(
+            filters=self.filters,
+            kernel_size=1,
+            strides=1,
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            padding="same",
+            data_format="channels_last",
+            use_bias=False,
+            name=self.name + "expand_conv",
+        )
+        self.bn1 = layers.BatchNormalization(
+            axis=BN_AXIS,
+            momentum=self.bn_momentum,
+            name=self.name + "expand_bn",
+        )
+        self.act = layers.Activation(self.activation, name=self.name + "activation")
+        self.depthwise = layers.DepthwiseConv2D(
+            kernel_size=self.kernel_size,
+            strides=self.strides,
+            depthwise_initializer=CONV_KERNEL_INITIALIZER,
+            padding="same",
+            data_format="channels_last",
+            use_bias=False,
+            name=self.name + "dwconv2",
+        )
+
+        self.bn2 = layers.BatchNormalization(
+            axis=BN_AXIS, momentum=self.bn_momentum, name=self.name + "bn"
+        )
+
+        self.se_conv1 = layers.Conv2D(
+            self.filters_se,
+            1,
+            padding="same",
+            activation=self.activation,
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            name=self.name + "se_reduce",
+        )
+
+        self.se_conv2 = layers.Conv2D(
+            self.filters,
+            1,
+            padding="same",
+            activation="sigmoid",
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            name=self.name + "se_expand",
+        )
+
+        self.output_conv = layers.Conv2D(
+            filters=self.output_filters,
+            kernel_size=1,
+            strides=1,
+            kernel_initializer=CONV_KERNEL_INITIALIZER,
+            padding="same",
+            data_format="channels_last",
+            use_bias=False,
+            name=self.name + "project_conv",
+        )
+
+        self.bn3 = layers.BatchNormalization(
+            axis=BN_AXIS, momentum=self.bn_momentum, name=self.name + "project_bn"
+        )
+
+    def build(self, input_shape):
+        if self.name is None:
+            self.name = backend.get_uid("block0")
+
+    def call(self, inputs):
+        # Expansion phase
+        if self.expand_ratio != 1:
+            x = self.conv1(inputs)
+            x = self.bn1(x)
+            x = self.act(x)
+        else:
+            x = inputs
+
+        # Depthwise conv
+        x = self.depthwise(x)
+        x = self.bn2(x)
+        x = self.act(x)
+
+        # Squeeze and excite
+        if 0 < self.se_ratio <= 1:
```

Comment:
Miss here, copied from the original implementation: https://github.com/keras-team/keras-cv/blob/5ea90682ffc2a5e28262d5e2261e8dd9f6bedeb2/keras_cv/models/efficientnet_v2.py#L637

The output phase is conditional, based on the `se_ratio`, which is totally off. Should've been unindented

Reply:
Thanks for catching this. This is clearly a mistake on my end during the
rewrite.

Upon inspection it seems this error never happens as MBConvBlock luckily
always has se_ratio equal to 0.25 which executes the final conv layer.

This should be absolutely fixed but it should not affect converted weights
nor existing models (here and in tf.keras.applications)

Reply:
Thanks for weighing in! I saw some `se_ratio`s being 0.0, but didn't check the conv type. All MBConv blocks had `se_ratio`>0! The FusedMBConv blocks didn't have this issue :)

This shouldn't affect the weight mapping or performance then. The PR should be ready for a merge if everything else looks okay @tanzhenyu